src/Microsoft.ML.PCA/PcaTransform.cs

```diff
@@ -98,40 +106,123 @@ public bool TryUnparse(StringBuilder sb)
             }
         }
 
+        public sealed class ColumnInfo
+        {
+            public readonly string Input;
+            public readonly string Output;
+            public readonly string WeightColumn;
```

Comment:
>WeightColumn [](start = 35, length = 12)

I'm not sure `Weight` is needed for the transformer

Reply:
Weights gets used in the Project() method, so evidently PCA can use a weight column. 

---
In reply to: [227068400](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227068400) [](ancestors = 227068400)